### PR TITLE
fix: move config watcher to graph command

### PR DESCRIPTION
**Pull request by anmonteiro**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Tentative fix for #4359

---

For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [ ] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

File: src/commands/dev/dev.js [hunk 1]
```diff
@@ -486,8 +511,10 @@ const createDevCommand = (program) => {
       'netlify dev',
       'netlify dev -d public',
       'netlify dev -c "hugo server -w" --targetPort 1313',
+      'netlify dev --graph',
```
Comment (anmonteiro) [inline src/commands/dev/dev.js:514]: drive-by adding a new example

File: src/commands/base-command.js [hunk 2]
```diff
@@ -437,27 +430,10 @@ class BaseCommand extends Command {
     const globalConfig = await getGlobalConfig()
     const { NetlifyAPI } = await jsClient
 
-    const configWatcher = new events.EventEmitter()
-
-    // Only set up a watcher if we know the config path.
-    if (configPath) {
```
Comment (erezrokah) [inline src/commands/base-command.js:443]: What do you think about keeping this in the base command, and only applying for `dev`?
I'm thinking we might want this for `functions:serve` and possibly other long running commands

File: src/commands/dev/dev.js (hunk 1 above)
Comment (erezrokah) [reply src/commands/dev/dev.js:514]: You'll need to run `npm run doc` to sync the docs with this change (and fix the CI)

File: src/commands/base-command.js (hunk 2 above)
Comment (anmonteiro) [reply src/commands/base-command.js:443]: I'd prefer to do that work when we need that for `functions:serve`. There are (as evidenced by this PR) wrinkles to figure out wrt memory usage and I believe exposing this in `BaseCommand` could give false confidence that this is ready for wide application.

**Review by sgrove — APPROVED** [review]
lgtm - thanks for helping out our users!
